client/data/emails/use-remove-email-forward-mutation.ts

```diff
@@ -27,24 +38,105 @@ export function useRemoveEmailForwardMutation(
 	const domainsQueryKey = getEmailDomainsQueryKey( selectedSiteId );
 
 	const suppliedOnSettled = mutationOptions.onSettled;
+	const suppliedOnMutate = mutationOptions.onMutate;
+	const suppliedOnError = mutationOptions.onError;
+	const suppliedOnSuccess = mutationOptions.onSuccess;
+
+	mutationOptions.mutationKey = MUTATION_KEY;
 
-	mutationOptions.onSettled = async ( data, error, variables, context ) => {
+	mutationOptions.onSettled = ( data, error, variables, context ) => {
 		suppliedOnSettled?.( data, error, variables, context );
 
-		await Promise.all( [
-			queryClient.invalidateQueries( emailDomainsQueryKey ),
-			queryClient.invalidateQueries( domainsQueryKey ),
-		] );
+		queryClient.invalidateQueries( emailDomainsQueryKey );
+		queryClient.invalidateQueries( domainsQueryKey );
 	};
 
-	return useMutation(
-		() =>
-			wp.req.get( {
-				path: `/domains/${ encodeURIComponent( domainName ) }/email/${ encodeURIComponent(
-					mailboxName
-				) }/delete`,
-				method: 'POST',
+	mutationOptions.onMutate = async ( emailForward ) => {
+		suppliedOnMutate?.( emailForward );
+
+		await queryClient.cancelQueries( emailDomainsQueryKey );
+
+		// Optimistically remove email forward from `useGetEmailAccountsQuery` data
+		const previousEmailAccountsQueryData = queryClient.getQueryData< any >( emailDomainsQueryKey );
+
+		const emailForwards = previousEmailAccountsQueryData.accounts?.[ 0 ]?.emails;
+
+		if ( emailForwards ) {
```

Comment:
It looks like this comment could be moved a few lines below where we actually remove that email forward:

```suggestion
		const previousEmailAccountsQueryData = queryClient.getQueryData< any >( emailDomainsQueryKey );

		const emailForwards = previousEmailAccountsQueryData.accounts?.[ 0 ]?.emails;

		// Optimistically remove email forward from `useGetEmailAccountsQuery` data
		if ( emailForwards ) {
```